src/snapred/backend/data/LocalDataService.py

```diff
@@ -535,292 +340,155 @@ def writeNormalizationRecord(
-        return record
+        indexor = self.calibrationIndexor(record.runNumber, record.useLiteMode)
+        version = indexor.thisOrNextVersion(version)
+        calibrationDataPath = indexor.versionPath(version)
+
+        # write the output d-spacing calibrated data
+        wsName = record.workspaces[wngt.DIFFCAL_OUTPUT]
+        ext = Config["calibration.diffraction.output.extension"]
+        filename = Path(wsName + ext)
+        self.writeRaggedWorkspace(calibrationDataPath, filename, wsName)
+
+        # write the diagnostic output
+        wsName = record.workspaces[wngt.DIFFCAL_DIAG]
+        ext = Config["calibration.diffraction.diagnostic.extension"]
+        filename = Path(wsName + ext)
+        self.writeWorkspace(calibrationDataPath, filename, wsName)
+
+        # write the diffcal table and attached masl
+        tableWSName = record.workspaces[wngt.DIFFCAL_TABLE]
+        maskWSName = record.workspaces[wngt.DIFFCAL_MASK]
+        ext = ".h5"
+        diffCalFilename = Path(tableWSName + ext)
+        self.writeDiffCalWorkspaces(
+            calibrationDataPath,
+            diffCalFilename,
+            tableWorkspaceName=tableWSName,
+            maskWorkspaceName=maskWSName,
+        )
 
     ##### REDUCTION METHODS #####
 
     @validate_call
     def readReductionRecord(self, runNumber: str, useLiteMode: bool, version: Optional[int] = None) -> ReductionRecord:
+        """
+        Will return a reduction record for the given version.
+        If no version given, will choose the latest applicable version from the index.
+        """
+        indexor = self.reductionIndexor(runNumber, useLiteMode)
         if version is None:
-            version = str(self._getLatestReductionVersionNumber(runNumber, useLiteMode))
-        record = None
-        if version is not None:
-            filePath: Path = self._constructReductionRecordFilePath(runNumber, useLiteMode, version)
-            with open(filePath, "r") as f:
-                record = ReductionRecord.model_validate_json(f.read())
-        return record
+            # NOTE Indexor.readRecord defaults to currentVersion
+            version = indexor.latestApplicableVersion(runNumber)
+        return indexor.readRecord(version)
 
-    def writeReductionRecord(self, record: ReductionRecord, version: Optional[int] = None) -> ReductionRecord:
+    def writeReductionRecord(self, record: ReductionRecord, version: Optional[int] = None):
         """
         Persists a `ReductionRecord` to either a new version folder, or overwrites a specific version.
         * side effect: updates version numbers of incoming `ReductionRecord`;
-        * must be called before any call to `writeReductionData`.
+        * side effect: updates version numbers of the `calculationParameters` on the record;
+        * side effect: creates the version directory if none exists;
         """
-        # For the moment, a single run number is assumed:
-        runNumber = record.runNumbers[0]
```

Comment:
`record.runNumber = record.runNumbers[0]` is automatically performed on reduction records

Reply:
Was it added as a `@property`?  (It possibly should be.  A `PrivateAttribute` or `Optional` attribute probably _not_.)